testgrid/tgrun/pkg/cli/run/queue.go

```diff
@@ -20,15 +20,17 @@ func QueueCmd() *cobra.Command {
 			v := viper.GetViper()
 
 			schedulerOptions := schedulertypes.SchedulerOptions{
-				APIEndpoint:  v.GetString("api-endpoint"),
-				APIToken:     v.GetString("api-token"),
-				OverwriteRef: v.GetBool("overwrite-ref"),
-				Ref:          v.GetString("ref"),
-				Staging:      v.GetBool("staging"),
-				Airgap:       v.GetBool("airgap"),
-				KurlVersion:  v.GetString("kurl-version"),
-				Spec:         v.GetString("spec"),
-				OSSpec:       v.GetString("os-spec"),
+				APIEndpoint:       v.GetString("api-endpoint"),
+				APIToken:          v.GetString("api-token"),
+				OverwriteRef:      v.GetBool("overwrite-ref"),
+				Ref:               v.GetString("ref"),
+				Staging:           v.GetBool("staging"),
+				Airgap:            v.GetBool("airgap"),
+				KurlVersion:       v.GetString("kurl-version"),
+				Spec:              v.GetString("spec"),
+				OSSpec:            v.GetString("os-spec"),
+				NumPrimaryNodes:   v.GetInt("num-primary-nodes"),
```

Comment:
can these be part of the Instance type? https://github.com/replicatedhq/kURL/blob/1100b3af5a2d0d16da9be413e52308e7ada32dfc/testgrid/tgrun/pkg/scheduler/types/types.go#L40-L50

Reply:
I guess it does make more sense to keep it with ``SchedulerOptions`` as running into a cluster with multiple nodes should be an option when we schedule a test, let me know what do you think?
and please refer to the end goal plan in this comment https://github.com/replicatedhq/kURL/pull/2853#discussion_r863313828

Reply:
when you queue a run you can test many combinations of specs with oses. im proposing we add the node counts to individual tests in the spec directly.

for example here https://github.com/replicatedhq/kURL/blob/main/testgrid/specs/full.yaml

if you add it as arguments to the cli queue command it will run that many nodes for each test in that spec but there may be specs where we only want to test single node installs